### ISSUE TITLE
Create and auth client using token rather than API key / password

### ISSUE THREAD

**Issue by eastlondoner** [pkgcloud/pkgcloud#352]
Currently Clients use tokens but it is not possible to create a Client using a token. I am using pkgcloud as part of a web application and I am not able to persist the Client objects between requests. Thus I have to create a new Client on every request with the user's username and password - which causes a call to the auth server on each request to my web app and I have to store the users credentials in memory / cookies or ask for them on every request.

If I could create a client from the auth token then I would be able to persist the auth token in my session state. 

An alternative for me would be to create a Client and pass in an Identity as they seem simple enough for me to persist in session state.

At least one person who is not me has asked about this on SO:
http://stackoverflow.com/questions/24182549/pkgcloud-token-authentication-with-openstack-for-many-clients

I am using Rackspace but I am pretty sure the same applies to OpenStack (if not others).

**Comment by rogerst8** [comment]
@kenperkins I know this is a little over a year late, but whatever happened with your prototype for creating a client with an auth token?

**Comment by chaddjohnson** [comment]
Yes, this needs to be made possible -- especially for those who wish to use this pack client-side with Browserify.

From http://sldn.softlayer.com/blog/waelriac/managing-softlayer-object-storage-through-rest-apis:

> To manage your object storage account, you can use a token for authentication. A token is usually valid for 24 hours and eliminates the need of using the credentials on every call.
> 
> You can obtain a token by passing the tenant:username/password to the object storage on your initial call. The tenant value is the username value retrieved in the above section. For example, our tenant value is ACC-278436-15, the username is the username used to log in to SoftLayer, and the password is the API-Key.

**Comment by saltukalakus** [comment]
Hi @kenperkins I have a similar problem. 

I am using express and passport. Luckily there is a passport plugin [passport-keystone](https://github.com/eddywashere/passport-keystone) which handles keystone authentication. After user is loged in passport stores the session which has openstack token and openstack service endpoints reachable from request object in the routes.

My question is, can I bypass  createClient creating a new interface, and just use my token and service endpoint list ?

This is actually quite a requirement if pkgcloud lib is used in a full featured web application.

**Comment by saltukalakus** [comment]
By the way passport-keystone uses pkgcloud. :+1:

**Comment by saltukalakus** [comment]
Hi @eastlondoner . How did you fixed your problem? Might be helpful also for me. regards

**Comment by eastlondoner** [comment]
It's a while since I worked on the project using this, which was later mothballed. But I think I just stored the user's credentials in memory and stuck in a TODO.
Which is obviously *bad*. Don't copy my example.
